docs/design/ovs-pipeline.md

```diff
@@ -333,58 +382,195 @@ more information on connection tracking in OVS.
+* `group:5` is used to set the target OVS group. Note that, the target group should be
+  created first before the flow is created.
+
+Dump the group entry with command `ovs-ofctl dump-groups br-int 5`, you may see the
+following:
+
+```text
+group_id=5,type=select,\
+bucket=bucket_id:0,weight:100,actions=load:0xa0a0002->NXM_NX_REG3[],load:0x23c1->NXM_NX_REG4[0..15],resubmit(,42),\
+bucket=bucket_id:1,weight:100,actions=load:0xa0a0003->NXM_NX_REG3[],load:0x23c1->NXM_NX_REG4[0..15],resubmit(,42),\
+bucket=bucket_id:2,weight:100,actions=load:0xa0a0004->NXM_NX_REG3[],load:0x23c1->NXM_NX_REG4[0..15],resubmit(,42)
+```
+
+There are three buckets which have the same weight. Every bucket has the same
+possibility to be selected since they have the same weight. The selected bucket
+will load Endpoint IP to NXM_NX_REG3, Endpoint port number to bits [0..15] in
+NXM_NX_REG4. Then the packet goes to table [EndpointDNATTable].
+
+If `service.spec.sessionAffinity` of the ClusterIP is "ClientIP", you may see the following flows:
+
+```text
+1. table=41, priority=200,tcp,reg4=0x10000/0x70000,nw_dst=10.107.100.231,tp_dst=443 actions=load:0x3->NXM_NX_REG4[16..18],load:0x1->NXM_NX_REG0[19],group:5
+2. table=41, priority=190,tcp,reg4=0x30000/0x70000,nw_dst=10.107.100.231,tp_dst=443 actions=\
+   learn(table=40,hard_timeout=300,priority=200,delete_learned,cookie=0x2040000000008, \
+     eth_type=0x800,nw_proto=6,NXM_OF_TCP_DST[],NXM_OF_IP_DST[],NXM_OF_IP_SRC[],\
+     load:NXM_NX_REG3[]->NXM_NX_REG3[],load:NXM_NX_REG4[0..15]->NXM_NX_REG4[0..15],load:0x2->NXM_NX_REG4[16..18],load:0x1->NXM_NX_REG0[19]),\
+   load:0x2->NXM_NX_REG4[16..18],\
+   resubmit(,42)
+```
 
-The table-miss flow entry (flow 2) for this table forwards all non-Service
-traffic to the next table, [EgressRuleTable].
+When a client (assumed that the source IP is 10.10.0.2) accesses the ClusterIP for the first time,
+the first packet will be matched by flow 1. Note that the action `load:0x3->NXM_NX_REG4[16..18]`,
+this indicates that the Service Endpoint selection result needs to be cached. Due
+to the same reason, the Endpoint selection has not really been done yet.
 
-In the future this table may support an additional mode of operations, in which
-it will implement kube-proxy functionality and take care of performing
-load-balancing / DNAT on traffic destined to Services.
+Dump the group entry with command `ovs-ofctl dump-groups br-int 5`, you can see the
+following:
+
+```text
+group_id=5,type=select,\
+bucket=bucket_id:0,weight:100,actions=load:0xa0a0002->NXM_NX_REG3[],load:0x23c1->NXM_NX_REG4[0..15],resubmit(,41),\
+bucket=bucket_id:1,weight:100,actions=load:0xa0a0003->NXM_NX_REG3[],load:0x23c1->NXM_NX_REG4[0..15],resubmit(,41),\
+bucket=bucket_id:2,weight:100,actions=load:0xa0a0004->NXM_NX_REG3[],load:0x23c1->NXM_NX_REG4[0..15],resubmit(,41)
+```
+
+Note the action `resubmit(,41)`, the first packet of the ClusterIP goes back to table
```

Comment:
", the first" -> "resubmits the first packet of a ClusterIP Service connection back to"

Reply:
Thanks, updated.